misk/src/test/kotlin/misk/hibernate/ReflectionQueryFactoryTest.kt

```diff
@@ -143,6 +143,69 @@ class ReflectionQueryFactoryTest {
     }
   }
 
+  @Test
+  fun inOperator() {
+    val m1 = NameAndReleaseDate("Rocky 1", LocalDate.of(2018, 1, 1))
+    val m2 = NameAndReleaseDate("Rocky 2", LocalDate.of(2018, 1, 2))
+    val m3 = NameAndReleaseDate("Rocky 3", LocalDate.of(2018, 1, 3))
+
+    transacter.transaction { session ->
+      session.save(DbMovie(m1.name, m1.releaseDate))
+      session.save(DbMovie(m2.name, m2.releaseDate))
+      session.save(DbMovie(m3.name, m3.releaseDate))
+
+      assertThat(queryFactory.newQuery<OperatorsMovieQuery>()
+          .releaseDateInVararg(m1.releaseDate, m3.releaseDate)
+          .listAs<NameAndReleaseDate>(session))
+          .containsExactly(m1, m3)
+
+      assertThat(queryFactory.newQuery<OperatorsMovieQuery>()
+          .releaseDateInCollection(listOf(m1.releaseDate, m3.releaseDate))
+          .listAs<NameAndReleaseDate>(session))
+          .containsExactly(m1, m3)
+    }
+  }
+
+  @Test
+  fun inOperatorWithNull() {
+    val m1 = NameAndReleaseDate("Rocky 1", LocalDate.of(2018, 1, 1))
+    val m2 = NameAndReleaseDate("Rocky 2", null)
+
+    transacter.transaction { session ->
+      session.save(DbMovie(m1.name, m1.releaseDate))
+      session.save(DbMovie(m2.name, m2.releaseDate))
+
+      assertThat(queryFactory.newQuery<OperatorsMovieQuery>()
+          .releaseDateInVararg(m1.releaseDate, null)
+          .listAs<NameAndReleaseDate>(session))
+          .containsExactly(m1)
+
+      assertThat(queryFactory.newQuery<OperatorsMovieQuery>()
+          .releaseDateInCollection(listOf(m1.releaseDate, null))
```

Comment:
so nulls are just filtered out? I don't see where that is happening

Reply:
I'm mostly just testing MySQL at this point. I think both with this and with the comparison operators it's worth writing documentation to warn how NULL will bite you in SQL.

I definitely don't want to change the semantics though; that makes mechanical porting extremely dangerous.

Reply:
ah got it.